proto/regen/ecocredit/v1/query.proto

```diff
@@ -559,11 +589,33 @@ message QueryCreditTypesResponse {
 // QueryParamsRequest is the Query/Params request type.
 message QueryParamsRequest {}
 
+// AllowedDenom represents the information for an allowed ask denom.
+message AllowedDenomInfo {
+  // denom is the bank denom to allow (ex. ibc/GLKHDSG423SGS)
+  string bank_denom = 1;
+
+  // display_denom is the denom to display to the user and is informational.
+  // Because the denom is likely an IBC denom, this should be chosen by
+  // governance to represent the consensus trusted name of the denom.
+  string display_denom = 2;
+
+  // exponent is the exponent that relates the denom to the display_denom and is
+  // informational
+  uint32 exponent = 3;
+}
```

Comment:
All the values provided in the `AllowedDenom` state message are already human-readable (i.e. there are no byte addresses that need to be converted to strings). Any reason for defining a separate message here?

Reply:
Initially I tried using `AllowedDenom` from the marketplace, but that is creating import cycles.

Reply:
Ah, ok. I see. Yea, this seems ok. We could also embed this within the `QueryParamsResponse` message given this is not native to the ecocredit.v1 package but we still need to define it separately. Ok to leave as is though.

Reply:
shouldn't this query actually go in `proto/regen/ecocredit/marketplace/v1`?

Reply:
It's the `Params` query that will be deprecated but `Params` is in `ecocredit.v1`.

Reply:
There is a separate query for the allowed denoms specifically but querying for params should return all params until we remove it.

Reply:
The main issue is if we have a query for params, it should return all params. The alternative is removing it and requiring users to use individual queries but keeping the params query for one more release seemed nice to have.

Reply:
I think this is ok for now. I can also clean up param proto messages a bit when updating the param names.